test/ProducerTraitTest.php

```diff
@@ -201,16 +208,18 @@ public function testEmitPendingPromiseThenFail() {
      * @expectedExceptionMessage The prior promise returned must resolve before invoking this method again
      */
     public function testDoubleAdvance() {
-        $this->producer->advance();
-        $this->producer->advance();
+        $iterator = $this->producer->iterate();
+        $iterator->advance();
+        $iterator->advance();
     }
 
     /**
      * @expectedException \Error
      * @expectedExceptionMessage Promise returned from advance() must resolve before calling this method
      */
     public function testGetCurrentBeforeAdvance() {
-        $this->producer->getCurrent();
+        $iterator = $this->producer->iterate();
+        $iterator->getCurrent();
```

Comment:
Why are these test changes necessary?

Reply:
That's a BC break we can't merge then.

Reply:
The change is in a trait marked internal. 😛